### PR TITLE
CSHARP-2757: Unable to call ToString on Projections.

### PR DESCRIPTION
**Pull request by DmitryLukyanov**
Evergreen: https://evergreen.mongodb.com/version/5d950c6fd6d80a086a00b0d9
Doc: https://docs.mongodb.com/manual/reference/operator/aggregation/toString/#behavior

### CODE REVIEW

File: src/MongoDB.Driver/Linq/Translators/AggregateLanguageTranslator.cs [hunk 1]
```diff
@@ -1276,5 +1314,19 @@ private bool TryTranslateSumResultOperator(PipelineExpression node, out BsonValu
             result = null;
             return false;
         }
+
+        private bool TryTranslateViaToString(MethodCallExpression node, BsonValue field, out BsonValue result)
```
Comment (DmitryLukyanov) [inline src/MongoDB.Driver/Linq/Translators/AggregateLanguageTranslator.cs:1323]: I've added the ability to specify not only decimal.ToString(), but all supported types for this command

File: src/MongoDB.Driver/Linq/Translators/AggregateLanguageTranslator.cs [hunk 2]
```diff
@@ -812,18 +832,16 @@ private bool TryTranslateDateTimeCall(MethodCallExpression node, out BsonValue r
...
                         {
                             { "format", format },
                             { "date", field }
                         });
-                        return true;
-                    }
-                    break;
+                    return true;
+                case "ToString": // without a format argument
+                    return TryTranslateViaToString(node, field, out result);
```
Comment (rstam) [inline src/MongoDB.Driver/Linq/Translators/AggregateLanguageTranslator.cs]: I don't think having two cases with the same name ("ToString") is helpful. I think the following more traditional code is easier to understand:

    switch (node.Method.Name)
    {
        case "ToString":
            if (node.Arguments.Count == 1)
            {
                var format = TranslateValue(node.Arguments[0]);
                result = new BsonDocument("$dateToString", new BsonDocument
                {
                    { "format", format },
                    { "date", field }
                });
                return true;
            }
            else
            {
                return TryTranslateViaToString(node, field, out result);
            }
    }

It's also a smaller diff which is often better.

**Review by rstam — CHANGES_REQUESTED** [review]
Looks mostly good to me with one minor comment.

If you prefer your version you can say so and not implement my comment.

File: src/MongoDB.Driver/Linq/Translators/AggregateLanguageTranslator.cs (hunk 2 above)
Comment (DmitryLukyanov) [reply src/MongoDB.Driver/Linq/Translators/AggregateLanguageTranslator.cs]: done

**Comment by DmitryLukyanov** [comment]
Evergreen:  https://evergreen.mongodb.com/version/5d9b59a357e85a2af21fb64a

Ready for review.

**Review by rstam — COMMENTED** [review]
LGTM

**Pull request closed without merging**